tracer/src/Datadog.Trace/Debugger/SpanCodeOrigin/EndpointDetector.cs

```diff
@@ -0,0 +1,296 @@
+// <copyright file="EndpointDetector.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Datadog.Trace.Pdb;
+using Datadog.Trace.VendoredMicrosoftCode.System.Collections.Immutable;
+using Datadog.Trace.VendoredMicrosoftCode.System.Reflection.Metadata;
+using Datadog.Trace.VendoredMicrosoftCode.System.Reflection.Metadata.Ecma335;
+
+namespace Datadog.Trace.Debugger.SpanCodeOrigin;
+
+internal static class EndpointDetector
```

Comment:
IMO, this `EndpointDetector` should have a bunch of unit tests. Right now I think we're relying on the `SpanCondOriginTests`, which is covering a lot more than just this. 

I think it would be easier to make sure we've covered all the variations and corner cases if we did that.

Reply:
unit tests15aefd2edeed4172cd83c45102e228e961e8aaeb